liquidprompt

```diff
@@ -929,6 +935,27 @@ __lp_theme_zsh_complete() {
     _describe 'theme' lp_theme_list
 }
 
+_lp_hash_color() {
+    # compute the hash of the argument and color it with the corresponding color in:
+    # (green, yellow, blue, purple, cyan)
+    # Red (would be index 1) is not used here, as it should be reserved for alerts.
+    local str=${1-}
+    local cksum="$(printf '%s' "$str" | cksum)"
+    local hash="${cksum%%[$' \t']*}"
+    __lp_foreground_color "$(( 2 + hash % 5 ))"
+    local color="${_LP_OPEN_ESC}${af_color}${_LP_CLOSE_ESC}"
```

Comment:
`__lp_foreground_color()` does not promise to return `$af_color`, that is why it is failing in the tests. It does this when color is disabled.

I think the correct fix is to check if `LP_ENABLE_COLOR` is enabled, or return right at the start of this function:
```bash
(( LP_ENABLE_COLOR )) || return 2
```
Obviously then this function will not always set `$lp_hash_color`, so you will need to protect where it is used.

Reply:
I was only partially right. Turns out there are still some bugs/quirks with how LP handles non-color environments that I need to work out. I can fix them in the future, but for now to prevent issues and make the tests pass, make this change
```suggestion
    local color="${_LP_OPEN_ESC}${af_color-}${_LP_CLOSE_ESC}"
```